src/operator.rs

```diff
@@ -72,6 +72,35 @@ macro_rules! declare_operator {
             }
         }
     };
+    ($category:literal [ $doc_url:literal ] : $ty:ident => $mongo_operator:literal [ $( $field:ident => $mongo_field:literal ),+ ]) => {
+        #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+        #[doc="["]
+        #[doc=$category]
+        #[doc="]("]
+        #[doc=$doc_url]
+        #[doc=") "]
+        #[doc="operator `"]
+        #[doc=$mongo_operator]
+        #[doc="`"]
+        #[allow(non_snake_case)]
+        pub struct $ty<$( $field ),+> where $( $field : Into<crate::mongo::bson::Bson> ),+ {
+            $( pub $field : $field ),+
+        }
+
+        impl<$( $field ),+> ::core::convert::From<$ty<$( $field ),+>> for crate::mongo::bson::Document where $( $field : Into<crate::mongo::bson::Bson> ),+ {
+            fn from(l: $ty<$( $field ),+>) -> crate::mongo::bson::Document {
+                crate::mongo::bson::doc! { $mongo_operator: {
```

Comment:
Not a big deal since it's only used inside this crate, but just in case it's best to use `$crate` in macros.
```suggestion
                $crate::mongo::bson::doc! { $mongo_operator: {
```